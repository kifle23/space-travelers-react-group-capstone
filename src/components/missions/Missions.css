table {
  font-family: arial, sans-serif;
  border-collapse: collapse;
  width: 100%;
}

td,
th {
  border: 1px solid hsl(180, 87%, 34%);
  text-align: left;
  padding: 8px;
}

tbody tr:nth-child(even) {
  background-color: #f2f2f2;
}

.mission-content {
  display: flex;
  flex-direction: column;
  row-gap: 40px;
  width: 100%;
  max-width: 1024px;
  margin-bottom: 60px;
}

.status-head {
  width: 100px;
}

.action-head {
  width: 130px;
}

p.member-status {
  font-size: 8px;
  background: aqua;
  padding: 5px;
  border: 1px solid aqua;
  border-radius: 5px;
  font-weight: bold;
  display: flex;
  justify-content: center;
}

p.not-member-status {
  font-size: 8px;
  background: #6c757e;
  padding: 5px;
  border: 1px solid #6c757e;
  border-radius: 5px;
  font-weight: bold;
  display: flex;
  justify-content: center;
  color: white;
}

button.join-btn {
  border: 1px solid hsl(180, 87%, 34%);
  padding: 8px;
  border-radius: 5px;
  background: transparent;
  font-size: 12px;
  color: #f68d36;
}

button.leave-btn {
  border: 1px solid red;
  padding: 8px;
  border-radius: 5px;
  background: transparent;
  color: red;
  font-size: 12px;
}
