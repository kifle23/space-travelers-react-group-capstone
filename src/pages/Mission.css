.mission-section {
  display: flex;
  justify-content: center;
  align-content: center;
  margin-top: 60px;
  flex-direction: column;
  align-items: center;
}
