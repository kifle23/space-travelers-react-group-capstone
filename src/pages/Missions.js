const Missions = () => (
  <div className="container"><h2>Missions Page</h2></div>
);

export default Missions;
