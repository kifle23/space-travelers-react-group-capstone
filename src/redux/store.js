import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './features/Rockets/RocketSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
  },
});

const selectRockets = (store) => store.rockets;
export { selectRockets };
export default store;
