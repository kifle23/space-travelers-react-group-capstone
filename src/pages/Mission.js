import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Missions from '../components/missions/Missions';
import {
  getMissionStatus,
  fetchMissions,
  getAllMissions,
} from '../redux/missionSlice';
import './Mission.css';

const Mission = () => {
  const dispatch = useDispatch();
  const { status } = useSelector(getMissionStatus);
  const missions = useSelector(getAllMissions);

  useEffect(() => {
    if (status === 'idle') {
      dispatch(fetchMissions());
    }
  }, [status, dispatch]);

  return (
    <div className="mission-section">
      <div className="mission-content">
        <table>
          <thead>
            <tr>
              <th>Mission</th>
              <th>Description</th>
              <th className="status-head">Status</th>
              <th className="action-head">Action</th>
            </tr>
          </thead>
          <tbody>
            {missions.map((x) => (
              <Missions key={x.id} mission={x} />
            ))}
          </tbody>
        </table>
      </div>
    </div>
  );
};

export default Mission;
