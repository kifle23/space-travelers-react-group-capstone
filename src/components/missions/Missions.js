import React from 'react';
import { useDispatch } from 'react-redux';
import { updateMission } from '../../redux/missionSlice';
import './Missions.css';

const Missions = (mission) => {
  const dispatch = useDispatch();

  const {
    mission: {
      id, missionName, description, reserved,
    },
  } = mission;

  const handleMission = () => {
    dispatch(updateMission(id));
  };

  return (
    <tr>
      <td>{missionName}</td>
      <td>{description}</td>
      <td>
        {reserved ? (
          <p className="not-member-status">ACTIVE MEMBER</p>
        ) : (
          <p className="member-status">NOT A MEMBER</p>
        )}
      </td>
      <td>
        {reserved ? (
          <button onClick={handleMission} type="button" className="leave-btn">
            LEAVE MISSION
          </button>
        ) : (
          <button onClick={handleMission} type="button" className="join-btn">
            JOIN MISSION
          </button>
        )}
      </td>
    </tr>
  );
};

export default Missions;
