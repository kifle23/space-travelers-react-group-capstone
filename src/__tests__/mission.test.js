import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/store';
import Mission from '../pages/Mission';
import Missions from '../components/missions/Missions';

describe('A test for missions component', () => {
  it('Check whether mission component renders', () => {
    const mission = {
      id: 'heloo',
      missionName: 'Commercial Resupply Services',
      description:
        'Commercial Resupply Services (CRS) are a series of contracts awarded by NASA from 2008â€“2016 for delivery of cargo and supplies to the International Space Station (ISS) on commercially operated spacecraft. The first CRS contracts were signed in 2008 and awarded $1.6 billion to SpaceX for 12 cargo transport missions and $1.9 billion to Orbital Sciences for 8 missions, covering deliveries to 2016. In 2015, NASA extended the Phase 1 contracts by ordering an additional three resupply flights from SpaceX and one from Orbital Sciences. After additional extensions late in 2015, SpaceX is currently scheduled to have a total of 20 missions and Orbital 10. SpaceX began flying resupply missions in 2012, using Dragon cargo spacecraft launched on Falcon 9 rockets from Space Launch Complex 40 at Cape Canaveral Air Force Station, Cape Canaveral, Florida. Orbital Sciences began deliveries in 2013 using Cygnus spacecraft launched on the Antares rocket from Launch Pad 0A at the Mid-Atlantic Regional Spaceport (MARS), Wallops Island, Virginia. A second phase of contracts (known as CRS2) were solicited and proposed in 2014. They were awarded in January 2016 to Orbital ATK, Sierra Nevada Corporation, and SpaceX, for cargo transport flights beginning in 2019 and expected to last through 2024.',
      reserved: false,
    };
    const renderedMission = renderer.create(
      <Provider store={store}>
        <Missions mission={mission} />
      </Provider>,
    );

    expect(renderedMission).toMatchSnapshot();
  });

  it('Check if it renders correctly', () => {
    const renderedMission = renderer.create(
      <Provider store={store}>
        <Mission />
      </Provider>,
    );

    expect(renderedMission).toMatchSnapshot();
  });
});
