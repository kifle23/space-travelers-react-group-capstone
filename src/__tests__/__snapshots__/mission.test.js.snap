// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A test for missions component Check if it renders correctly 1`] = `
<div
  className="mission-section"
>
  <div
    className="mission-content"
  >
    <table>
      <thead>
        <tr>
          <th>
            Mission
          </th>
          <th>
            Description
          </th>
          <th
            className="status-head"
          >
            Status
          </th>
          <th
            className="action-head"
          >
            Action
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;

exports[`A test for missions component Check whether mission component renders 1`] = `
<tr>
  <td>
    Commercial Resupply Services
  </td>
  <td>
    Commercial Resupply Services (CRS) are a series of contracts awarded by NASA from 2008â€“2016 for delivery of cargo and supplies to the International Space Station (ISS) on commercially operated spacecraft. The first CRS contracts were signed in 2008 and awarded $1.6 billion to SpaceX for 12 cargo transport missions and $1.9 billion to Orbital Sciences for 8 missions, covering deliveries to 2016. In 2015, NASA extended the Phase 1 contracts by ordering an additional three resupply flights from SpaceX and one from Orbital Sciences. After additional extensions late in 2015, SpaceX is currently scheduled to have a total of 20 missions and Orbital 10. SpaceX began flying resupply missions in 2012, using Dragon cargo spacecraft launched on Falcon 9 rockets from Space Launch Complex 40 at Cape Canaveral Air Force Station, Cape Canaveral, Florida. Orbital Sciences began deliveries in 2013 using Cygnus spacecraft launched on the Antares rocket from Launch Pad 0A at the Mid-Atlantic Regional Spaceport (MARS), Wallops Island, Virginia. A second phase of contracts (known as CRS2) were solicited and proposed in 2014. They were awarded in January 2016 to Orbital ATK, Sierra Nevada Corporation, and SpaceX, for cargo transport flights beginning in 2019 and expected to last through 2024.
  </td>
  <td>
    <p
      className="member-status"
    >
      NOT A MEMBER
    </p>
  </td>
  <td>
    <button
      className="join-btn"
      onClick={[Function]}
      type="button"
    >
      JOIN MISSION
    </button>
  </td>
</tr>
`;
